src/appleseed/foundation/utility/statistics.h

```diff
@@ -431,6 +452,46 @@ const T* Statistics::Entry::cast(const Entry* entry)
 }
 
 
+//
+// Statistics::PercentEntry class implementation.
+//
+
+template<typename T>
+Statistics::PercentEntry<T>::PercentEntry(
+    const std::string&          name,
+    const T&                    numerator,
+    const T&                    denominator,
+    const std::streamsize       precision)
+  : Entry(name)
+  , m_numerator(numerator)
+  , m_denominator(denominator)
+  , m_precision(precision)
+{
+}
+
+template <typename T>
+std::unique_ptr<Statistics::Entry> Statistics::PercentEntry<T>::clone() const
+{
+    return std::unique_ptr<Statistics::Entry>(new PercentEntry(*this));
+}
+
+template <typename T>
+void Statistics::PercentEntry<T>::merge(const Entry* other)
+{
+    const PercentEntry* pother = cast<PercentEntry>(other);
+
+    m_numerator = m_numerator * pother->m_denominator + pother->m_numerator * m_denominator;
```

Comment:
The problem of this approach is that each time entries are merged, numerator and denominator are getting larger. Ideally we would simplify the fraction afterward (divide by greatest common divisors).

Whether it'll work reliably as is depends on the width of type `T`. What `T` do we have in practice?

Reply:
I'm using size_t in the adaptive sampler, but I should take a look on other uses. Also, I could use the final ratio in only 1 variable and call `pretty_ratio`. But i'm not sure how to add values if we don't keep the denominator and numerator

Reply:
Well you need to keep the numerator and denominator obviously, you just need to simplify the fraction after every merge. I don't think it's necessary in practice since we're not merging millions of values together.